src/EditorFeatures/CSharpTest2/Recommendations/RecommenderTests.cs

```diff
@@ -27,10 +24,10 @@ public abstract class RecommenderTests : TestBase
     {
         protected static readonly CSharpParseOptions CSharp9ParseOptions = CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp9);
 
-        protected string keywordText;
-        internal Func<int, CSharpSyntaxContext, Task<ImmutableArray<RecommendedKeyword>>> RecommendKeywordsAsync;
+        protected string? keywordText;
```

Comment:
I don't think this should be nullable, all implementors of `RecommenderTests` must set this. perhaps an abstract property `protected abstract string KeywordText {get;}` would be better?

Reply:
Sounds great, but `NativeIntegerKeywordRecommenderTests` is using this hack to test for 2 different keywords:
https://github.com/dotnet/roslyn/blob/2cf85e445dab552c3e176de5d736464aa696d37f/src/EditorFeatures/CSharpTest2/Recommendations/NativeIntegerKeywordRecommenderTests.cs#L25-L45

So it disables our ability to do abstraction thing (I believe, unit tests don't work if the test is in abstract class). I can make separate tests for `nint` and `nuint`, but it will cause code duplication, which is not great either.

So there are 2 cases: we can leave it without changes, but with nullable `keywordText`, which is not how it should be or use abstraction to avoid unintended nullability, but cause code duplication. What variant to take?

Reply:
Please just break that into two sets of recommender tests.  the tests can be in the base abstract class.  teh derived will only set one.

Reply:
Keyword recommender defines keyword text on fly using reflection:
https://github.com/dotnet/roslyn/blob/2cf85e445dab552c3e176de5d736464aa696d37f/src/EditorFeatures/CSharpTest2/Recommendations/KeywordRecommenderTests.cs#L43-L56
I don't understand how to reimplement this one without need to explicitly override `KeywordText` in all derived classes